stdlib/public/core/ArrayBuffer.swift

```diff
@@ -567,6 +567,86 @@ extension _ArrayBuffer {
     }
   }
 
+  @inlinable @_alwaysEmitIntoClient
+  static var associationKey: UnsafeRawPointer {
+    //We never dereference this, we just need an address to use as a unique key
+    UnsafeRawPointer(Builtin.addressof(&_swiftEmptyArrayStorage))
+  }
+  
+  @inlinable @_alwaysEmitIntoClient
+  internal func getAssociatedBuffer() -> _ContiguousArrayBuffer<Element>? {
+    let getter = objc_getAssociatedObject as @convention(c)(
+      Any,
+      UnsafeRawPointer
+    ) -> Any?
+    let typedGetter = unsafeBitCast(
+      getter,
+      to: (@convention(c)(
+        AnyObject,
+        UnsafeRawPointer
+      ) -> Unmanaged<AnyObject>?).self
```

Comment:
Using Unmanaged here prevents swiftc from generating pointless calls to objc_retainAutoreleasedReturnValue